mem.go

```diff
@@ -43,12 +50,12 @@ func (m *MemoryStore) GetAccessLimit(ctx context.Context, projectID uint64, cycl
 	if !ok {
 		return nil, proto.ErrAccessKeyNotFound
 	}
-	return limit, nil
+	return &limit, nil
 }
 
 func (m *MemoryStore) SetAccessCycle(ctx context.Context, projectID uint64, cycle *proto.Cycle) error {
 	m.Lock()
-	m.cycles[projectID] = cycle
+	m.cycles[projectID] = *cycle
```

Comment:
Maybe we can pass the value to function itslef already dereferenced, or add a nil check.
```suggestion
	m.cycles[projectID] = cycle
```